iotqatools/cb_ngsiv2_utils.py

```diff
@@ -0,0 +1,371 @@
+            assert False, 'ERROR: [NETWORK ERROR] {}'.format(e)
+
+        # Log data
+        PqaTools.log_fullRequest(comp='CB', response=response, params=parameters)
+
+        return response
+
+    def version(self):
+        """
+        Get CB version
+        """
+        url = self.default_endpoint + self.path_version
+
+        # send the request for the subscription
+        response = self.__send_request('get', url, self.headers)
+        return response
+
+    def statistics(self):
+        """
+        Get CB statistics
+        """
+        url = self.default_endpoint + self.path_statistics
+
+        # send the request for the subscription
+        response = self.__send_request('get', url, self.headers)
+        return response
+
+    def create_entity(self, headers, payload, params=None):
+        """
+        Create a entity in ContextBroker with the standard entity creation
+        :param headers: headers for the requests (fiware-service, fiware-servic-path and x-auth-token)
+        :param payload: the payload
+        :param params: params of the query if applicable
+        The payload has to be like:
+            {
+              "type": "Room",
+              "id": "Bcn-Welt",
+              "temperature": {
+                "value": 21.7
+              },
+              "humidity": {
+                "value": 60
+              },
+              "location": {
+                "value": "41.3763726, 2.1864475",
+                "type": "geo:point",
+                "metadata": {
+                  "crs": {
+                    "value": "WGS84"
+                  }
+                }
+              }
+            }
+        """
+        headers.update(self.headers)
+        headers.update({'content-type': 'application/json'})
+        return self.__send_request('post', self.path_create_entity, payload=payload, headers=headers, query=params,
+                                   verify=None)
+
+    def list_entities(self, headers, filters=None):
+        """
+        Retrieves a list of entities which match different criteria (by id, idPattern, type or those which match a
+        query or geographical query)
+        :param headers:
+        :param filters:
+        :rtype : object
+        :
+
+        """
+
+        # Add default headers to the request
+        headers.update(self.headers)
+
+        # Set the filters of the requests as params
+        if filters is not None:
+            params = filters
+        else:
+            params = None
+
+        return self.__send_request('get', self.path_list_entities, headers=headers, verify=None, query=params)
+
+
+if __name__ == '__main__':
+    cb = CbNgsi10v2Utils('127.0.0.1', 'http')
+    md = MetadataV2('crs', 'WGS84')
+    md2 = MetadataV2('crs2', 'WGS83')
+    print(md.get_metadata())
+    attr = AttributeV2('location', '41.3763726, 2.1864475', att_type='geo:point')
```

Comment:
In order to add more info about this sample usage please do not omit the params name in the funcs.
For instance:
`attr2 = AttributeV2(att_name='temperature', att_value=21.7) `